cloudify_kubernetes/workflows.py

```diff
@@ -0,0 +1,82 @@
+from cloudify.decorators import workflow
+from cloudify.exceptions import NonRecoverableError
+from cloudify.workflows import ctx
+
+from fabric.api import env, run, settings
+import json
+import re
+
+
+FORBIDDEN_CHARACTERS = '$!&`|><;#{}()'
+
+AGENT_CONFIG_PROPERTY = 'agent_config'
+AGENT_CONFIG_USER_PROPERTY = 'user'
+AGENT_CONFIG_KEY_PROPERTY = 'key'
+IP_PROPERTY = 'ip'
+
+
+def _retrieve_ssh_credentials(node):
```

Comment:
It should first look for agent_config in runtime properties and then check the node properties.